src/gpu/generic/sycl/binary_kernels.hpp

```diff
@@ -88,10 +89,15 @@ struct binary_kernel_vec_t {
                 dims[i] = 1;
                 strides[i] = INT_MAX;
             }
+            if (i < src0_md().ndims()) {
+                is_same_tag = is_same_tag
+                        && (src0_md().strides()[i] == src1_md().strides()[i]);
```

Comment:
There's a `similar_to` function in `memory_desc_wrapper`, which is more reliable as it checks way more than just strides.

Reply:
In this specific case, we only need to check the strides. I was looking at how to call the `similar_to` function inside the kernel and will need to make several changes to do this specifically. Let me know if I am missing something else here. Thanks.